velero-plugins/common/shared.go

```diff
@@ -8,13 +8,30 @@ import (
 	"strings"
 
 	"github.com/fusor/ocp-velero-plugin/velero-plugins/clients"
+	"github.com/openshift/library-go/pkg/image/reference"
+	"github.com/sirupsen/logrus"
 	"k8s.io/apimachinery/pkg/api/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/version"
 )
 
-func GetRegistryInfo(major, minor string) (string, error) {
+func GetRegistryInfo(major, minor string, log logrus.FieldLogger) (string, error) {
```

Comment:
So this returns the registry clusterIP and bound port e.g. `1.1.1.1:5000`

Reply:
The old behavior (before this PR) was to use clusterIP:port for 3.x, and pull from the ConfigMaps for 4.x. The issue is that this is broken functionality if the 3.x cluster has a configured internal registry hostname. As of this PR, this is now only the backup behavior -- the first thing we look at is pulling ImageStreams, just like `oc registry info` does. I kept the old method as a fallback because it's possible for the imagestream check to fail, but it should really be an edge case.